enterprise-search/utils/retriever.py

```diff
@@ -0,0 +1,107 @@
+"""Retriever wrapper for Google Cloud Enterprise Search."""
+from __future__ import annotations
+
+import json
+from typing import Any, Dict, List, Optional
+
+from google.protobuf.json_format import MessageToDict
+from google.cloud.discoveryengine_v1beta  import SearchServiceClient, SearchRequest
+
+from pydantic import BaseModel, Extra, root_validator
+
+from langchain.schema import BaseRetriever, Document
+from langchain.utils import get_from_dict_or_env
+
+class EnterpriseSearchRetriever(BaseRetriever, BaseModel):
+    """Wrapper around Google Cloud Enterprise Search."""
+    client: Any = None #: :meta private:
+    serving_config: Any = None #: :meta private:Any
+    content_search_spec: Any = None #: :meta private:Any
+    project_id: str = None
+    search_engine_id: str = None
+    serving_config_id: str = 'default_config'
+    location_id: str = 'global'
+    max_snippet_count: int = 3
+    credentials: Any = None
+    "The default custom credentials (google.auth.credentials.Credentials) to use "
+    "when making API calls. If not provided, credentials will be ascertained from "
+    "the environment."
+
+
+    class Config:
+        """Configuration for this pydantic object."""
+        extra = Extra.forbid
+        arbitrary_types_allowed = True
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        try:
+            from google.cloud import discoveryengine_v1beta
+        except ImportError:
+            raise ImportError(
+                "google.cloud.discoveryengine is not installed. "
+                "Please install it with pip install google-cloud-discoveryengine"
+            )
+
+        project_id = get_from_dict_or_env(values, "project_id", "PROJECT_ID")
+        values["project_id"] = project_id
+        search_engine_id = get_from_dict_or_env(values, "search_engine_id", "SEARCH_ENGINE_ID")
+        values["search_engine_id"] = search_engine_id
+        location_id = get_from_dict_or_env(values, "location_id", "LOCATION_ID")
+        values["location_id"] = location_id
+        max_snippet_count = get_from_dict_or_env(values, "max_snippet_count", "MAX_SNIPPET_COUNT")
+        values["max_snippet_count"] = max_snippet_count
```

Comment:
This feels like a lot of extra variables that might not be needed. Since `EnterpriseSearchRetriever` has fields for most or all of these elements, would it make sense to just fill those with `self.project_id` etc?